### ISSUE TITLE
Is is possible to launch apps supporting startup notifications with sn_rules/partially at least right from command line

### ISSUE THREAD

**Issue by xor512** [awesomeWM/awesome#2526]
This is more of a question then an issue. As I already know that for programs (like xfce4-terminal or gvim) that support startup notifications it is possible to use awful.spawn function and pass it sn_rules with staff like:

    awful.spawn("gvim", {
            floating = true,
            screen = 1,
            tag = "fair5",
            maximized_vertical   = false,
            maximized_horizontal = false
    })

saying how it should launch and where.

This is very cool but is there a way to specify similar stuff right from command line? Foe example.

- I'm on a fair tag with 4 xfc4-tertminals. Now I want to gvim xxx.txt. But I don't want it to become part of the fair layout. I want to launch it as floating (without manually pressing Crtrl-Mod-Space after it loads)
- I'm on a tile tag with 2 xfce4-terminals (left's say the tag is is 3rd one). Now I want to launch mardkown reader like retext in floatinb mode on float tag which is the 2nd one. I know I can launch it, then make floating with Ctrll-Mod-Space and then press Ctrl-Mod-2 to move to float tag.

But Is there a way to specify such stuff directly from command-line? If it is verbose I can add aliases to ~/.basrc.

Thanks!

**Comment by psychon** [comment]
How exactly do you launch things? Is this from a terminal? From a key combination controlled by awesome?

> But Is there a way to specify such stuff directly from command-line?

How about using `awesome-client` for this? `awesome-client 'require("awful.spawn")("gvim", { floating = true })'`

**Comment by psychon** [comment]
> is it OK to require just one function an this will return a function itself so that it can be called write away?

Well, no. In general, things are a bit complicated. It is just that pretty much everything directly below `awful` is its own module and can be required directly. You can of course just also do `require("awful").spawn`. It should not make much of a difference.

**Comment by Elv13** [comment]
> I cannot see how can I specify an exact tag within this.

The `tyrannical` module allows to codify the rules for this, so awesome does that for me. There is a bunch of extra tag properties like `intrusive` and `new_tag` to handle the other use cases. Also, my built-in launcher has support for some tags. See the top and bottom wibars (sorry for the lack of contrast and colors)

![screenshot_20190101_153914](https://user-images.githubusercontent.com/340384/50576237-654c2a00-0ddb-11e9-8b5c-e370edaac3dc.png)

**Comment by xor512** [comment]
No, it's cool. It shows how much can be done with awesome. I'm just using it much like mate/xfce4/lxde/gnome3 with QtCreator on a float tag, jumping to tile tag to launch binaries and stuff after there and occasionaly showing admins what the problem is with this funny fair tag with 4 terminals. But even with that I don't understand how people can use Gnome3 and all the new stuff. I do understand Gnome2, xfce, lxde, WIndows even, but not Gnome3. It's so much like Max OS X/iPhone and the fonts and all the corners are so much NOT like Mac OS X/iPhone that this is just pathetic.

**Comment by actionless** [comment]
> and the fonts and all the corners

you probably one of those guys who judging a car by color :-)

**Comment by actionless** [comment]
> Nah

yes: you clearly saying what gnome is sucking because they chosen default font and rounding of default themes not like on nyanyabook or whatever

**Comment by xor512** [comment]
No,

I say it is sucking because they aim to be like Mac OS X, fail in that and do nothing apart of that.

**Comment by actionless** [comment]
so far you just didn't mentioned how exactly it copying macos and how else they failed to copy it except for missing the fonts and border radius

it just a bit annoying what when people discuss some distribution or DE on news portals they instead of discussing technical points are discussing default choices of the themes of the distro packagers

and if such talks are agreeable on news portals, it's a bit so for me to them here on sorta technical resource for devs

**Comment by xor512** [comment]
- The feature wich Ctrl-arrows is cool. Up is makes it maximised, left - right go to the left half, right half, just like tiled WM, cool
- Is there anything else there? Grouping my terminals into one f... btn on the left where I'm not able to distinguish them one from another is stupid. Same for grouping any taskbar icons in this (this was disablebale from Windows)
- Do they allow to remember workspace configuration and rules that Spotify/Netflix whatever this is "Entertainment" let's go to the far far away workspace names "Entertainment" (and so on)
- Do they allow to open windows from command line exactly specifying where they should go
- The super feature is cool is like Alt-F2, but awesome has Mod-P and even dmenu had all that

Where is the new stuff I can use? It is either I'm blind and cannot see and cannot cook it right and just don't know or I just don't want to be it Mac OS X as I've used Mac OS X (well Hackintosh) for about and year, and no, I don't want it to use it any more.

How technical a conversation about WM (it's what you see!) can be? It is all about how it looks like and how it feels like.

**Comment by xor512** [comment]
PS On the other hand maybe I've went too far. Sorry for that.

**Comment by xor512** [comment]
I still want it. Imagine my work machine with 2 monitors and my home machine with 1 monitor. What I can do is swap rc.lua to a 1 monitor one and run ctrl-mod-r (restart awesome). I'm afraid to do it still today because I'm afraid everything will end up on the wrong monitor and in rdesktop I won't see anything.

So what I do (I'm seeing only 1 monitor) is that I "pkill firefox", "pkill qtcreator" and such and recreate them on the tag on the 1 of the monitors which I can see as the only one from rdesktop. Having your code I can once save what I've done and when I rdesktop next it can move my firefox and qtcreator (launched already) without a need to restart them. Can it? How can I use it for this?

**Comment by actionless** [comment]
thing about 2 displays <-> 1 display what you're talking about sounds like this issue: https://github.com/awesomeWM/awesome/issues/1382

**Comment by xor512** [comment]
Thnks. Please give me a day or smth to figure these links. As it is tiresome to constantly kill and relaunch smth. The worst is that after I come to work from home the launched stuff is also lost somewhere. Like in another session. But it is not an another session. Part of the stuff from 2-montors I can see with 1-monitor when I rdesktop. Probably each time I rdesktop I should change rc.lua and restart.

**Comment by actionless** [comment]
all multidisplay bugs which you saying i think related to the changes related to handling clients then display disappears (like dis/connecting external screen or when display goes disconnected because of sleep/hibernate of the machine), i think in #1382 and other related issues @psychon was doing some suggestions on that
